sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalTypescriptSensor.java

```diff
@@ -119,24 +123,60 @@ private void analyze(
       LOG.debug(String.format("Analyzing %s typescript file(s) with the following configuration file %s", inputFilesForThisConfig.size(), tsconfigPath));
 
       SonarTSRunnerCommand command = executableBundle.getSonarTsRunnerCommand(tsconfigPath, inputFilesForThisConfig, typeScriptRules);
-      SonarTSRunnerResponse[] responses = executeExternalRunner(command, localTypescript);
-
-      for (SonarTSRunnerResponse response : responses) {
-        FileSystem fileSystem = sensorContext.fileSystem();
-        InputFile inputFile = fileSystem.inputFile(fileSystem.predicates().hasAbsolutePath(response.filepath));
-        if (inputFile != null) {
-          saveHighlights(sensorContext, response.highlights, inputFile);
-          saveSymbols(sensorContext, response.symbols, inputFile);
-          saveMetrics(sensorContext, response, inputFile);
-          saveCpd(sensorContext, response.cpdTokens, inputFile);
-          saveFailures(sensorContext, response.issues, typeScriptRules);
-        } else {
-          LOG.error("Failed to find input file for path `" + response.filepath + "`");
+      if (checkNodeVersion(executableBundle.getNodeExecutable())) {
+        SonarTSRunnerResponse[] responses = executeExternalRunner(command, localTypescript);
+
+        for (SonarTSRunnerResponse response : responses) {
+          FileSystem fileSystem = sensorContext.fileSystem();
+          InputFile inputFile = fileSystem.inputFile(fileSystem.predicates().hasAbsolutePath(response.filepath));
+          if (inputFile != null) {
+            saveHighlights(sensorContext, response.highlights, inputFile);
+            saveSymbols(sensorContext, response.symbols, inputFile);
+            saveMetrics(sensorContext, response, inputFile);
+            saveCpd(sensorContext, response.cpdTokens, inputFile);
+            saveFailures(sensorContext, response.issues, typeScriptRules);
+          } else {
+            LOG.error("Failed to find input file for path `" + response.filepath + "`");
+          }
         }
       }
     }
   }
 
+  private static boolean checkNodeVersion(String nodeExecutable) {
+    LOG.debug("Checking node version");
+    String version = getNodeVersion(nodeExecutable);
+    if (version == null) {
+      return false;
+    }
+
+    Pattern versionPattern = Pattern.compile("v?(\\d+)\\.\\d+\\.\\d+");
+    Matcher versionMatcher = versionPattern.matcher(version);
+    if (versionMatcher.matches()) {
+      int major = Integer.parseInt(versionMatcher.group(1));
+      if (major < MIN_NODE_VERSION) {
+        LOG.error(String.format("Only Node.js v%s or later is supported, got %s", MIN_NODE_VERSION, version));
+        return false;
+      }
+    } else {
+      LOG.error(String.format("Failed to parse Node.js version, got %s", version));
+      return false;
+    }
+
+    LOG.debug(String.format("Using Node.js %s", version));
+    return true;
+  }
+
+  private static @Nullable String getNodeVersion(String nodeExecutable) {
```

Comment:
@stas-vilchik we usually put `@Nullable` on separate line in front of the method